_posts/2021-04-28-#17331.md

```diff
@@ -0,0 +1,111 @@
+---
+layout: pr
+date: 2021-04-28
+title: "Use effective values throughout coin selection"
+pr: 17331
+authors: [achow101]
+components: ["wallet"]
+host: Xekyo
+status: upcoming
+commit: 4ac1add
+---
+
+## Notes
+
+- Wallets maintain their balance as a pool of Unspent Transaction Outputs
+  (UTXOs).
+
+- _Coin selection_ refers to the process of picking UTXOs from the wallet's UTXO
+  pool to fund a transaction.
+
+- Beyond the primary goal of funding a transaction, the secondary goals of coin
+  selection are:
+
+    - Minimizing short term and long term fees:  
+      We want to both pay the least right now, but we also want to be good
+      stewards of our UTXO pool and not rack up large UTXO counts that will
+      translate to huge future transaction fees.
+
+    - Maintaining financial privacy:  
+      Ideally, our transactions leak as little information about ourselves and
+      our business partners as possible.
+
+    - Reliably move the payment(s) to finalization:  
+      Reduce the chance of transactions being conflicted and transfer value in a
+      limited time-frame
+
+- Using unconfirmed inputs can make transactions unreliable. When a foreign
+  input is used, the parent transaction may time out or get replaced. Even using
+  self-sent unconfirmed funds may delay the new transaction if the parent
+  transaction has an extensive ancestry, is extraordinarily large, or used a
+  lower feerate than targeted for the child transaction.
```

Comment:
This could be moved to the "Reliably move the payments to finalization" section above!

Reply:
Great idea, done!